src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -924,10 +919,44 @@ private void ProcessInterlacedDefilteredScanline<TPixel>(ReadOnlySpan<byte> defi
         }
     }
 
+    /// <summary>
+    /// Decodes and assigns the color palette to the metadata
+    /// </summary>
+    /// <param name="palette">The palette buffer.</param>
+    /// <param name="alpha">The alpha palette buffer.</param>
+    /// <param name="pngMetadata">The png metadata.</param>
+    private static void AssignColorPalette(ReadOnlySpan<byte> palette, ReadOnlySpan<byte> alpha, PngMetadata pngMetadata)
+    {
+        if (palette.Length == 0)
+        {
+            return;
+        }
+
+        Color[] colorTable = new Color[palette.Length / Unsafe.SizeOf<Rgb24>()];
+        ReadOnlySpan<Rgb24> rgbTable = MemoryMarshal.Cast<byte, Rgb24>(palette);
+        for (int i = 0; i < colorTable.Length; i++)
+        {
+            colorTable[i] = new Color(rgbTable[i]);
+        }
```

Comment:
Nit: I see several loops like this one in codecs now. We can maybe define a `public static Color.FromPixel<TPixel>(Span<TPixel>, Span<Color>)`  analogous to the bulk `ToPixel<TPixel>(...)` method.

Reply:
Without knowing the component precision of the `TPixel` we don't know which boxed pixel to assign to. We could do it in v4 though if we added additional information to `PixelTypeInfo` https://github.com/SixLabors/ImageSharp/discussions/2534#discussioncomment-7171432

Reply:
Ah yeah this indeed goes well beyond this PR, nevermind then.